src/Features/Core/Portable/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -28,24 +28,41 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TSimpleArgumentSyntax, TA
             public async Task ComputeRefactoringsAsync(
                 CodeRefactoringContext context, SyntaxNode root, CancellationToken cancellationToken)
             {
-                if (context.Span.Length > 0)
-                {
-                    return;
-                }
+                var document = context.Document;
+                var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
 
-                var position = context.Span.Start;
-                var token = root.FindToken(position);
-                if (token.Span.Start == position &&
-                    IsCloseParenOrComma(token))
+                var argument = await refactoringHelperService.TryGetSelectedNodeAsync<TSimpleArgumentSyntax>(document, context.Span, cancellationToken).ConfigureAwait(false);
+                if (argument == null && context.Span.IsEmpty)
                 {
-                    token = token.GetPreviousToken();
-                    if (token.Span.End != position)
+                    // Pre-helper implementation allowed cursor being anywhere inside the arg. expression as long as 
```

Comment:
Nit: The first paragraph here seems confusing for anyone who does not have context about the helpers that you mention. May be it is fine just to have second paragraph in the comment here?